### ISSUE TITLE
Nan Loss 

### ISSUE THREAD

**Issue by qqc222111** [lppllppl920/EndoscopyDepthEstimation-Pytorch#37]
![image](https://github.com/lppllppl920/EndoscopyDepthEstimation-Pytorch/assets/152779319/2154bef3-f2cc-4ede-bada-983286dbc4d4)
why is the number 0.9 in ”_bilinear_interpolate(img_masks, u_2_flat, v_2_flat) * img_masks >= 0.9“. Can it be replaced it in order to avoid causing the nan loss value, and what should it  be replaced with? 

![image](https://github.com/lppllppl920/EndoscopyDepthEstimation-Pytorch/assets/152779319/39482250-da84-4c6b-8877-ce533948c48e)
When I replace it with 0.24, it can run but the loss_depth_consistency is too small compared to loss_sparse_flow. Is it normal?

![image-20231203183544161](https://github.com/lppllppl920/EndoscopyDepthEstimation-Pytorch/assets/152779319/b1dafc59-ebcf-4260-8415-a5152d458ce2)
and in the function "torch.nn.functional.grid_sample(input=im.permute(0, 3, 1, 2), grid=grid, mode='bilinear',
                                           align_corners=True, padding_mode=padding_mode).permute(0, 2, 3, 1)" in _bilinear_interpolate(), 
why the result is out of (0,1), as the the value of im is either 0 or 1.

**Comment by qqc222111** [comment]
Now the problem above is nearly solved by adjusting pre_process_data. I remain the number 0.9. The problem is that the loss_depth_consistency is still too small compared to loss_sparse_flow. Is it normal?
![image](https://github.com/lppllppl920/EndoscopyDepthEstimation-Pytorch/assets/152779319/a7d494be-43d9-4cad-8ff6-4ded3adfde99)

**Comment by qqc222111** [comment]
and the image is not trained completely.

**Comment by joelive** [comment]
> Now the problem above is nearly solved by adjusting pre_process_data. I remain the number 0.9. The problem is that the loss_depth_consistency is still too small compared to loss_sparse_flow. Is it normal?

how do you adjusting pre_process_data to make loss_depth_consistency normal?
